src/components/atoms/SelectInput/SelectInput.model.ts

```diff
@@ -0,0 +1,10 @@
+export interface ISingleOption {
+  id: number;
+  value: string;
+}
+
+export interface ISelectInput {
+  isMulti: boolean;
+  options: ISingleOption[];
+  handleOnClick: (callback: () => ISingleOption[]) => void | undefined;
```

Comment:
Instead of undefined, we can use optional type

```suggestion
  handleOnClick?: (callback: () => ISingleOption[]) => void;
```